matchzoo/engine/base_model.py

```diff
@@ -1,207 +1,107 @@
-    def fixed_hyper_parameters(self, config):
-        """Set fixed hyper parameters."""
-        for key, value in config.items():
-            self._fixed_hyper_parameters[key] = value
-            self._list_fixed_hyper_parameters.append(key)
+    def __init__(self,
+                 params: typing.Optional[BaseModelParams] = None,
+                 backend: keras.models.Model = None):
+        self._params = params or self.get_default_params()
+        self._backend = backend
 
-    @property
-    def default_hyper_parameters(self):
-        """Universal parameters that can be use across varies models."""
-        return self._default_hyper_parameters
-
-    @default_hyper_parameters.setter
-    def default_hyper_parameters(self, config):
-        """Set default hyper parameters."""
-        allowed_default_parameters = self._default_hyper_parameters.keys()
-        for key, value in config.items():
-            if key not in allowed_default_parameters:
-                raise ValueError(
-                    '{} not in allowed default parameters: {}.'.format(
-                        key,
-                        allowed_default_parameters))
-            self._default_hyper_parameters[key] = value
+    @classmethod
+    def get_default_params(cls) -> BaseModelParams:
+        return BaseModelParams()
 
     @property
-    def model_specific_hyper_parameters(self):
-        """Get model specific hyper parameters."""
-        return self._model_specific_hyper_parameters
-
-    @model_specific_hyper_parameters.setter
-    def model_specific_hyper_parameters(self, config):
-        """Set model specific hyper parameters."""
-        self._model_specific_hyper_parameters = config
+    def params(self) -> BaseModelParams:
+        return self._params
 
     @property
-    def user_given_parameters(self):
-        """Get user given hyper parameters."""
-        return self._user_given_parameters
-
-    def _aggregate_hyper_parameters(self):
-        """Merge all of the hyper parameters.
-
-        This function is used internaly.`
-
-        # Returns:
-            conf: final model  configuration.
-        """
-        conf = self._default_hyper_parameters.copy()
-        # Universal config will be overwrite by model_specific.
-        # Then merge.
-        conf.update(self._model_specific_hyper_parameters)
-        # Merged universal config and model config will be.
-        # overriten by user_given_parameters.
-        conf.update(self._user_given_parameters)
-        # Filter fixed-parameters from configuration.
-        [conf.pop(key, None) for key in self._list_fixed_hyper_parameters]
-        # Merge default fixed parameters.
-        conf.update(self._fixed_hyper_parameters)
-        return conf
+    def backend(self) -> keras.models.Model:
+        return self._backend
 
     @abc.abstractmethod
-    def _build(self):
-        """Build model, each sub class need to impelemnt this method.
+    def build(self):
+        """"""
```

Comment:
so basically, you didn't write any docstring  and delete my docstring.

Reply:
I deleted pretty much the whole thing and rewrote it. I agree that docstring is important, but making progress and speed up the review cycle is also important. @bwanglzu we have talked about this and please let the others know the plan. So far I have 100% on the `base_model.py` so that's worth reviewing.

Reply:
我只是疑惑这个函数，一样的功能。

Reply:
Well, since I'm rewrote a lot, I wasn't sure that the old docstrings are still accurate. It's better to have no docstrings than misleading docstrings. If it's still accurate once you guys are good with the implementation, I will move back appropriate docstrings you wrote in the old version.

Reply:
okay